pkgs/applications/networking/instant-messengers/nheko/default.nix

```diff
@@ -1,70 +1,72 @@
-{ lib, stdenv, fetchFromGitHub
-, cmake, cmark, lmdb, mkDerivation, qtbase, qtmacextras
-, qtmultimedia, qttools, mtxclient, boost, spdlog, olm, pkgconfig
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, cmark
+, lmdb
+, lmdbxx
+, tweeny
+, mkDerivation
+, qtbase
+, qtmacextras
+, qtmultimedia
+, qttools
+, qtquickcontrols2
+, qtgraphicaleffects
+, mtxclient
+, boost17x
```

Comment:
I think you should keep `boost` here and call the package with `boost = boost17x` in all-packages.nix.

Reply:
Some people like it this way, and some people like it as I did :/ whenever I do either I'm commented... I don't have a strong opinion on this repetitive dilemma but there's no convention that I'm aware of. Maybe @rnhmjoj you could spare me please another cycle of Git checkout -> edit -> ... Git commit -> Git push -f ?

Reply:
Uhm, you're right: I checked with openssl and boost (two major sources of incompatible versions) and it seems we are really split 50-50 on this topic. I think the version should be treaded like an override and kept out of the nix expression of a package but given there is no unanimity let's have it your way.